front/src/modules/trainschedule/components/Timetable/TimetableTrainCard.tsx

```diff
@@ -91,6 +101,25 @@ function TimetableTrainCard({
     deleteTrainScheduleById({ id: train.id })
       .unwrap()
       .then(() => {
+        const updatedSimulation = (simulation.trains as SimulationReport[]).filter(
+          (simulationTrain) => simulationTrain.id !== train.id
+        );
+        if (updatedSimulation.length > 0) {
+          const trainWithSamePath = updatedSimulation.find(
+            (simulationTrain) => simulationTrain.path === train.path_id
+          );
+          // If there isn't any train left with the same path and the deleted
+          // train is the projected one, we want to refetch all trains
+          if (!trainWithSamePath && projectionPathIsUsed) {
+            setTrainResultsToFetch(undefined);
+          } else {
+            // We don't fetch anything
+            setTrainResultsToFetch([]);
+          }
+        } else {
+          setTrainResultsToFetch(undefined);
+        }
+        dispatch(updateSimulation({ trains: updatedSimulation }));
```

Comment:
`remainingTrains` is more descriptive than `updatedSimulation`.

I suggest you make the logic here a little less verbose.
```suggestion
        const remainingTrains = (simulation.trains as SimulationReport[]).filter(
          (simulationTrain) => simulationTrain.id !== train.id
        );
        
        let trainResultsToFetch = undefined; // by default we assume that we need to refetch all
        
        if (remainingTrains.length > 0 && projectionPathIsUsed) {
          const trainWithSamePath = updatedSimulation.find(
            (simulationTrain) => simulationTrain.path === train.path_id
          );
          // If there is at least one remaining train following the route of the one that was deleted
          if (trainWithSamePath ) {
            trainResultsToFetch = []; //No need to refetch
          }
        }
        dispatch(updateSimulation({ trains: remainingTrains }));
```

Reply:
Done (adapted your suggestion a little)